Spec/Auth.swift

```diff
@@ -637,9 +637,124 @@ class Auth : QuickSpec {
+                    }
+                }
+
+                it("query will provide a TokenDetails") {
+                    guard let testTokenDetails = getTestTokenDetails(clientId: "tester") else {
+                        fail("TokenDetails is empty")
+                        return
+                    }
+
+                    let encoder = ARTJsonLikeEncoder()
+                    encoder.delegate = ARTJsonEncoder()
+                    guard let jsonTokenDetails = encoder.encodeTokenDetails(testTokenDetails) else {
+                        fail("Invalid TokenDetails")
+                        return
+                    }
+
+                    let options = ARTClientOptions()
+                    options.authUrl = NSURL(string: "http://echo.ably.io")
+                    expect(options.authUrl).toNot(beNil())
+                    // JSON with TokenDetails
+                    options.authParams = [NSURLQueryItem]()
+                    options.authParams?.append(NSURLQueryItem(name: "type", value: "json"))
+                    options.authParams?.append(NSURLQueryItem(name: "body", value: jsonTokenDetails.toUTF8String))
+
+                    let rest = ARTRest(options: options)
+                    rest.httpExecutor = testHTTPExecutor
+
+                    waitUntil(timeout: testTimeout) { done in
+                        rest.auth.requestToken(nil, withOptions: nil, callback: { tokenDetails, error in
+                            expect(testHTTPExecutor.requests.last?.URL?.host).to(equal("echo.ably.io"))
+                            expect(error).to(beNil())
+                            expect(tokenDetails).toNot(beNil())
+                            expect(tokenDetails?.clientId) == testTokenDetails.clientId
+                            expect(tokenDetails?.capability) == testTokenDetails.capability
+                            expect(tokenDetails?.issued).toNot(beNil())
+                            expect(tokenDetails?.expires).toNot(beNil())
+                            if let issued = tokenDetails?.issued, let testIssued = testTokenDetails.issued {
+                                expect(issued.compare(testIssued)) == NSComparisonResult.OrderedSame
+                            }
+                            if let expires = tokenDetails?.expires, let testExpires = testTokenDetails.expires {
+                                expect(expires.compare(testExpires)) == NSComparisonResult.OrderedSame
+                            }
+                            done()
+                        })
+                    }
+                }
+
+                it("query will provide a TokenRequest") {
+                    var tokenRequest: ARTTokenRequest?
+                    waitUntil(timeout: testTimeout) { done in
+                        // Sandbox and valid TokenRequest
+                        ARTRest(options: AblyTests.commonAppSetup()).auth.createTokenRequest(nil, options: nil, callback: { newTokenRequest, error in
```

Comment:
Shouldn't we add something to the token request (capability, etc.) and then check in the retrieved `TokenDetails` that it matches? As is, the library could be ignoring the token request from `authUrl` and we wouldn't notice.